packages/pigeon/lib/ast.dart

```diff
@@ -82,6 +74,30 @@ class Api extends Node {
   }
 }
 
+/// A parameter to a generic entity.
+class TypeArgument {
+  /// Constructor for [TypeArgument].
+  TypeArgument({
+    required this.dataType,
+    required this.isNullable,
+    this.typeArguments,
+  });
+
+  /// A string representation of the base datatype.
+  final String dataType;
+
+  /// The type arguments to [TypeArgument].
```

Comment:
This is pretty hard to parse; maybe adding an example " (e.g., `String` for `List<String>`)" would be helpful.

Reply:
Done, added the example to the class docstring and reworded this one.